src/tree/registries/Azure/AzureRegistryDataProvider.ts

```diff
@@ -0,0 +1,218 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import type { Registry as AcrRegistry, RegistryListCredentialsResult } from '@azure/arm-containerregistry';
+import { AzureSubscription, VSCodeAzureSubscriptionProvider } from '@microsoft/vscode-azext-azureauth';
+import { RegistryV2DataProvider, V2Registry, V2RegistryItem, V2Repository, V2Tag, registryV2Request } from '@microsoft/vscode-docker-registries';
+import { CommonRegistryItem, isRegistryRoot } from '@microsoft/vscode-docker-registries/lib/clients/Common/models';
+import * as vscode from 'vscode';
+import { createAzureContainerRegistryClient, getResourceGroupFromId } from '../../../utils/azureUtils';
+import { ACROAuthProvider } from './ACROAuthProvider';
+
+export interface AzureRegistryItem extends V2RegistryItem {
+    readonly subscription: AzureSubscription;
+    readonly id: string;
+}
+
+export interface AzureSubscriptionRegistryItem extends CommonRegistryItem {
+    readonly subscription: AzureSubscription;
+    readonly type: 'azuresubscription';
+}
+
+export type AzureRegistry = V2Registry & AzureRegistryItem & {
+    readonly registryProperties: AcrRegistry;
+};
+
+export type AzureRepository = V2Repository;
+
+export type AzureTag = V2Tag;
+
+export function isAzureSubscriptionRegistryItem(item: unknown): item is AzureSubscriptionRegistryItem {
+    return !!item && typeof item === 'object' && (item as AzureSubscriptionRegistryItem).type === 'azuresubscription';
+}
+
+export function isAzureRegistry(item: unknown): item is AzureRegistry {
+    return !!item && typeof item === 'object' && (item as AzureRegistryItem).additionalContextValues?.includes('azureContainerRegistry');
+}
+
+export function isAzureRepository(item: unknown): item is AzureRepository {
+    return !!item && typeof item === 'object' && (item as AzureRepository).additionalContextValues?.includes('azureContainerRepository');
+}
+
+export function isAzureTag(item: unknown): item is AzureTag {
+    return !!item && typeof item === 'object' && (item as AzureTag).additionalContextValues?.includes('azureContainerTag');

```

Comment:
These could be simplified to something like this:
`return isCommonTag(item) && item.additionalContextValues?.includes('azureContainerTag')`

Reply:
agreed

Reply:
with the context value changes, this can just be return isCommonTag(item) && item.additionalContextValues?.includes('azure')